stonesoup/reader/video.py

```diff
@@ -151,50 +152,72 @@ class FFmpegVideoStreamReader(UrlReader, FrameReader):
     .. _ffmpeg-python: https://github.com/kkroening/ffmpeg-python
     .. _FFmpeg: https://www.ffmpeg.org/download.html
 
-    """  # noqa:E501
-
-    buffer_size = Property(int,
-                           doc="Size of the frame buffer. The frame "
-                               "buffer is used to cache frames in cases "
-                               "where the stream generates frames faster "
-                               "than they are ingested by the reader. "
-                               "If `buffer_size` is less than or equal to "
-                               "zero, the buffer size is infinite.",
-                           default=1)
-
-    input_opts = Property(dict,
-                          doc="FFmpeg input options, provided in the form of "
-                              "a dictionary, whose keys correspond to option "
-                              "names. (e.g. ``{'fflags': 'nobuffer'}``). "
-                              "The default is ``{}``.",
-                          default={})
-    output_opts = Property(dict,
-                           doc="FFmpeg output options, provided in the form "
-                               "of a dictionary, whose keys correspond to "
-                               "option names. The default is "
-                               "``{'f': 'rawvideo', 'pix_fmt': 'rgb24'}``.",
-                           default={'f': 'rawvideo', 'pix_fmt': 'rgb24'})
+    """
+
+    input_file: str = Property(
+        doc="Input source to read video stream from, passed as input file argument. This can "
+            "include any valid FFmpeg input e.g. rtsp URL, device name when using 'dshow'/'v4l2'")
+    buffer_size: int = Property(
+        default=1,
+        doc="Size of the frame buffer. The frame buffer is used to cache frames in cases where "
+            "the stream generates frames faster than they are ingested by the reader. If "
+            "`buffer_size` is less than or equal to zero, the buffer size is infinite.")
+    input_opts: Mapping[str, str] = Property(
+        default=None,
+        doc="FFmpeg input options, provided in the form of a dictionary, whose keys correspond to "
+            "option names. (e.g. ``{'fflags': 'nobuffer'}``). The default is ``{}``.")
+    output_opts: Mapping[str, str] = Property(
+        default=None,
+        doc="FFmpeg output options, provided in the form of a dictionary, whose keys correspond "
+            "to option names. The default is ``{'f': 'rawvideo', 'pix_fmt': 'rgb24'}``.")
+    filters: Sequence[Tuple[str, Sequence[str], Mapping[str, str]]] = Property(
```

Comment:
Here we are declaring the 2nd element of the `Tuple` as `Sequence[str]`, although the code will work fine when provided with `Tuple[int, int]`. 

As I am not really accustomed to the typing package, I am just wondering whether this can be expressed differently, e.g. `Sequence[any]`? This assumed that `Tuple[int, int]` is a valid subtype...

Reply:
Yes, `Tuple` is a sub type of `Sequence`, so that can be used. Looking at FFMpeg code, it converts any inputs to strings, so `Sequence[Any]` would work. This is also the same for kwargs, where key and value are converted, so I'll change that as well to `Mapping[Any, Any]`.